### ISSUE TITLE
Docker hub eosio/eos@v1.3.1 does not work with Docker/docker-compose.yml

### ISSUE THREAD

**Issue by jcalfee** [EOSIO/eos#5905]
Docker hub eosio/eos@v1.3.1 does not work with [Docker/docker-compose.yml](https://github.com/EOSIO/eos/blob/master/Docker/docker-compose.yml).

To test, modify the docker-compose file to use the docker image instead of the source code build:
```bash
Docker/docker-compose.yml
@@ -1,15 +1,8 @@
 version: "3"
 
 services:
-  builder:
-    build:
-      context: builder
-    image: eosio/builder
-
   nodeosd:
-    build:
-      context: .
-    image: eosio/eos
+    image: eosio/eos:v1.3.1
     command: /opt/eosio/bin/nodeosd.sh --data-dir /opt/eosio/bin/data-dir -e --http-alias=nodeosd:8888 --http-alias=127.0.0.1:8888 --http-alias=localhost:8888
     hostname: nodeosd
     ports:
@@ -24,7 +17,7 @@ services:
     stop_grace_period: 10m
 
   keosd:
-    image: eosio/eos
+    image: eosio/eos:v1.3.1
     command: /opt/eosio/bin/keosd --wallet-dir /opt/eosio/bin/data-dir --http-server-address=127.0.0.1:8900 --http-alias=keosd:8900 --http-alias=localhost:8900
     hostname: keosd
     links:
```
This is the results:

```bash
$ docker-compose up
Recreating docker_nodeosd_1 ... 
WARNING: Service "nodeosd" is using volume "/opt/eosio/bin/data-dir" from the previous container. Host mapping "nodeos-data-volume" has no effect. Remove the existing containers (with `docker-compose rm nodeosd`) to use the host volume mapRecreating docker_nodeosd_1 ... done
Recreating docker_keosd_1   ... 
Recreating docker_keosd_1   ... done
Attaching to docker_nodeosd_1, docker_keosd_1
nodeosd_1  | 
nodeosd_1  | 
keosd_1    | 2018-10-04T14:30:37.876 thread-0   wallet_plugin.cpp:42          plugin_initialize    ] initializing wallet plugin
keosd_1    | 2018-10-04T14:30:37.876 thread-0   http_plugin.cpp:422           plugin_initialize    ] configured http to listen on 127.0.0.1:8900
nodeosd_1  | 2018-10-04T14:30:36.738 thread-0   main.cpp:136                  main                 ] Throw location unknown (consider using BOOST_THROW_EXCEPTION)
nodeosd_1  | Dynamic exception type: boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::program_options::unknown_option> >
nodeosd_1  | std::exception::what: unrecognised option 'wallet-dir'
nodeosd_1  | 
keosd_1    | 2018-10-04T14:30:37.876 thread-0   http_plugin.cpp:554           add_handler          ] add api url: /v1/keosd/stop
keosd_1    | 2018-10-04T14:30:37.876 thread-0   http_plugin.cpp:486           plugin_startup       ] start listening for http requests
keosd_1    | 2018-10-04T14:30:37.876 thread-0   wallet_api_plugin.cpp:73      plugin_startup       ] starting wallet_api_plugin
keosd_1    | 2018-10-04T14:30:37.876 thread-0   http_plugin.cpp:554           add_handler          ] add api url: /v1/wallet/create
keosd_1    | 2018-10-04T14:30:37.877 thread-0   http_plugin.cpp:554           add_handler          ] add api url: /v1/wallet/create_key
docker_nodeosd_1 exited with code 254
```

Use to work in v1.2 ..

**Comment by ericiles** [comment]
Looks like the docker-compose file is fine. The issue appears to be related to the config.ini being used. I will determine which options changed with this release, and update the config.ini accordingly.

**Comment by conr2d** [comment]
configuration options for `keosd` are moved into `eosio-wallet/config.ini`

According to release note of v1.3.0:
**Unknown configuration items in config.ini will now error on startup (#5678)**
Prior to 1.3.0 unknown configuration items in config.ini would be silently ignored. This behavior has been changed in 1.3.0 to an error that prevents startup of nodeos. Users with old auto-generated config.ini files may need to remove some options that are no longer available in nodeos for example wallet-dir or unlock-timeout

**Comment by hama** [comment]
edit which config.ini ?

**Comment by elbowlicker** [comment]
I have the same problem. Did you solve this problem?

**Comment by elbowlicker** [comment]
> edit which config.ini ?



> edit which config.ini ?
I have the same problem. Did you solve this problem?

**Comment by jgiszczak** [comment]
The updated config.ini was released in v1.4.0, though v1.4.0 should not be used because of a defect.  Use v1.4.1, which has a valid config.ini for nodeos running in Docker.
